openerp/tools/translate.py

```diff
@@ -742,18 +742,20 @@ def get_root_view(xml_id):
                 _logger.error("name error in %s: %s", xml_name, str(exc))
                 continue
             objmodel = registry.get(obj.model)
-            if (objmodel is None or field_name not in objmodel._columns
+            if (objmodel is None or (
+                        field_name not in objmodel._columns and
+                        field_name not in objmodel._fields)
```

Comment:
Old API columns are automatically mapped to new API fields at construction [by the metaclass](https://github.com/odoo/odoo/blob/880e951f344cc1a4f96040b5e9e7d07936ac0d81/openerp/models.py#L250-L254), delegating the instantiation to the [`to_field()` method of the columns](https://github.com/odoo/odoo/blob/880e951f344cc1a4f96040b5e9e7d07936ac0d81/openerp/osv/fields.py#L191-L219). The converse is true only for stored fields indeed.

So @ajalbut is right, looking in `_fields` should be sufficient.

It would simplify a bit the rest of the patch and ensure that `field_def` is always a new API `field`.